conans/client/source.py

```diff
@@ -23,23 +23,16 @@ def retrieve_exports_sources(remote_manager, cache, recipe_layout, conanfile, re
         return None
 
     if conanfile.exports_sources is None and not hasattr(conanfile, "export_sources"):
-        mkdir(export_sources_folder)
         return None
 
-    # TODO: cache2.0 check what behaviour for this in 2.0
-    # If not path to sources exists, we have a problem, at least an empty folder
-    # should be there
-    current_remote = cache.get_remote(recipe_layout.reference)
-    if current_remote:
-        current_remote = remotes[current_remote]
-    if not current_remote:
+    if not remote:
```

Comment:
Probably the ``retrieve_export_sources`` shouldn't be tied to the current remote. If this happens in a ``conan upload`` command, it will use the remote you are uploading to, but instead the sources must be downloaded from other remote that contains this revision.

Reply:
I have checked, this is being called in ``uploader`` with the same remote that you are going to upload it, so this would fail there (not sure why it is not failing in the test suite now). We need to check this.

Reply:
I removed the remote argument. Now if there's some remote selected it's taken and otherwise it searches for the first remote with sources to download them.